app/scripts/views/behaviors/settings.js

```diff
@@ -34,9 +34,18 @@ define(function (require, exports, module) {
             if (options.success) {
               success = options.success;
             }
-            if (options.endpoint) {
-              endpoint = options.endpoint;
+
+            // Check the `redirectTo` param sent from server, if it matches
+            // a known path redirect there, else just go to settings.
+            const redirectTo = view.getSearchParam('redirectTo');
```

Comment:
Interesting, I wasn't aware that there was a `/settings?setting=<setting name>` ability. I originally opted to use the `redirectTo` param because most of the infrastructure to handle it was already in place in the auth-server and js-client.

I'll see if there is a reasonable way to consolidate these.